Schema Rework

BoogieMonster1O1 opened the pull request:
Condenses most fields into other objects to allow for expansion since codecs can only have sixteen fields. 
Also renamed `dimensions` to `worlds` as it's more clearer.

BoogieMonster1O1 committed:
Create a separate codec for conditions and properties

 Changes to be committe
	modified:   settings.gradle
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/MonoColorSkybox.java
	new file:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/RotatableSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/textured/AnimatedSquareTexturedSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/textured/SquareTexturedSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/textured/TexturedSkybox.java
	new file:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/Conditions.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/Decorations.java
	new file:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/DefaultProperties.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/HeightEntry.java

AMereBagatelle commented:
Debatable that it's clearer to change `dimensions` to `worlds` unless that's what Mojang calls them now or something.  To a mod developer, yes it makes more sense, but from a player standpoint it seems like it might be confusing.

BoogieMonster1O1 committed:
Yeet tests

 Changes to be committed:
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/textured/SquareTexturedSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/textured/TexturedSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/Rotation.java
	deleted:    src/testmod/java/io/github/amerebagatelle/fabricskyboxestest/DevelopmentTests.java
	modified:   src/testmod/resources/fabric.mod.json

BoogieMonster1O1 commented:
> Debatable that it's clearer to change `dimensions` to `worlds` unless that's what Mojang calls them now or something. To a mod developer, yes it makes more sense, but from a player standpoint it seems like it might be confusing.

The concept of a dimension isn't usually what it is.
a 'dimension' can either be a dimension type, or a world. Plain dimensions don't exist in 1.16 as they did in 1.15. 
also, people will understand. it's not that hard once you've read the docs

BoogieMonster1O1 committed:
Convert codecs

 Changes to be committed:
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/MonoColorSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/textured/AnimatedSquareTexturedSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/textured/SquareTexturedSkybox.java

BoogieMonster1O1 committed:
Imports

 Changes to be committed:
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/MonoColorSkybox.java

BoogieMonster1O1 commented:
Almost complete. 
Need to finish writing docs and tests

FlashyReese reviewed (approved):
Apart from the `dimension` or `world` renaming, LGTM.

BoogieMonster1O1 committed:
Optional fields

 Changes to be committed:
	modified:   docs/schema-v2.md
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/SkyboxManager.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/MonoColorSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/textured/AnimatedSquareTexturedSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/textured/SquareTexturedSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/Conditions.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/DefaultProperties.java
	new file:   src/testmod/java/io/github/amerebagatelle/fabricskyboxes/Test.java

BoogieMonster1O1 committed:
Start docs

 Changes to be committed:
	modified:   docs/schema-v2.md
	new file:   docs/schema-v2.old.md
	modified:   example/sky_template_basic.json
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/RGBA.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/Rotation.java

AMereBagatelle commented on docs/schema-v2.old.md:
```diff
@@ -0,0 +1,160 @@
+# Skybox specification
```
Considering I never released anything with this schema version, there's no need to have a old version

BoogieMonster1O1 replied:
I’ve just kept that for my reference. 
was planning to clean up after I finish the docs

BoogieMonster1O1 committed:
List -> vector

 Changes to be committed:
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/textured/TexturedSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/Rotation.java

BoogieMonster1O1 committed:
remove constructor :crab:

 Changes to be committed:
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/mixin/skybox/WorldRendererAccess.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/textured/TexturedSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/Decorations.java

BoogieMonster1O1 committed:
Two more

 Changes to be committed:
	modified:   docs/schema-v2.md
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/HeightEntry.java

BoogieMonster1O1 committed:
Textures, weather

 Changes to be committed:
	modified:   docs/schema-v2.md
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/Weather.java

BoogieMonster1O1 committed:
Decorations, Conditions

 Changes to be committed:
	modified:   docs/schema-v2.md
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/SkyboxManager.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/Decorations.java

BoogieMonster1O1 committed:
Default properties

 Changes to be committed:
	modified:   docs/schema-v2.md
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/Rotation.java

BoogieMonster1O1 committed:
Finish writing docs

 Changes to be committed:
	modified:   docs/schema-v2.md
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/MonoColorSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/textured/AnimatedSquareTexturedSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/textured/SquareTexturedSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/Decorations.java
	modified:   src/testmod/resources/fabric.mod.json

BoogieMonster1O1 committed:
Tests

 Changes to be committed:
	modified:   docs/schema-v2.md
	modified:   src/testmod/java/io/github/amerebagatelle/fabricskyboxes/Test.java

BoogieMonster1O1 committed:
pretty printing

 Changes to be committed:
	modified:   src/testmod/java/io/github/amerebagatelle/fabricskyboxes/Test.java

BoogieMonster1O1 committed:
Merge remote-tracking branch 'up/1.16.x-dev' into schema-rework

BoogieMonster1O1 committed:
Resource packs

 Changes to be committed:
	modified:   .gitignore
	new file:   run/resourcepacks/animated_square_textured/assets/animated_square_textured/sky/animated_square_textured.json
	new file:   run/resourcepacks/animated_square_textured/pack.mcmeta
	new file:   run/resourcepacks/monocolor/assets/animated_square_textured/sky/monocolor.json
	new file:   run/resourcepacks/monocolor/pack.mcmeta
	new file:   run/resourcepacks/square_textured/assets/square_textured/sky/square_textured.json
	new file:   run/resourcepacks/square_textured/pack.mcmeta
	modified:   src/testmod/java/io/github/amerebagatelle/fabricskyboxes/Test.java
	deleted:    src/testmod/resources/assets/fabricskyboxes/sky/test_animated_textured.json
	deleted:    src/testmod/resources/assets/fabricskyboxes/sky/test_mono.json.txt
	deleted:    src/testmod/resources/assets/fabricskyboxes/sky/test_texured_v2.json
	modified:   src/testmod/resources/fabric.mod.json

BoogieMonster1O1 committed:
Fields

BoogieMonster1O1 committed:
Fix type

AMereBagatelle commented on src/main/java/io/github/amerebagatelle/fabricskyboxes/SkyboxManager.java:
```diff
@@ -1,16 +1,17 @@
 import java.util.Objects;
 import java.util.function.Supplier;
 
+import com.google.common.collect.Lists;
```
Accidental cleanup code run?

BoogieMonster1O1 replied:
🤔

AMereBagatelle commented on src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java:
```diff
@@ -319,37 +323,29 @@ public void parseJson(JsonObjectWrapper jsonObjectWrapper) {
+        }
+    }
+
+    private void processIds(JsonElement element, List<Identifier> list) {
```
Umm, maybe I'm misunderstanding here, but wouldn't this just add the element to a copied list, not the actual list?

BoogieMonster1O1 replied:
This code was duplicated in some methods. Just decided to make a method for it. 
And no, lists are passed by reference, which allows for stuff like this to happen.

AMereBagatelle reviewed (changes requested):
Other than cleanup things that I assume you are already planning to take care of, here's the things I found

BoogieMonster1O1 committed:
Multiple namespaces

 Changes to be committed:
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/resource/SkyboxResourceLoader.java

BoogieMonster1O1 committed:
remove old schema doc :crab:

BoogieMonster1O1 commented:
everything works now. i've also added the ability to use multiple namespaces cos why not

AMereBagatelle commented:
Still not sure on the test skyboxes changes... imo should be reverted back to the testmod as it's just simpler

BoogieMonster1O1 commented:
> Still not sure on the test skyboxes changes... imo should be reverted back to the testmod as it's just simpler

it allows easily enabling and disabling skyboxes without having to move files around, reload classes, and reload resources. and since debugging is slower than running normally, it makes sense. the testmod is still there because I need it for the generator

AMereBagatelle commented:
I see.  It's fine then.

AMereBagatelle commented:
If you are done, I will merge the PR.

BoogieMonster1O1 commented:
finished it yep
please review the docs. i feel that there could be some improvements though i dont know where

AMereBagatelle commented on docs/schema-v2.md:
````diff
@@ -45,19 +69,121 @@ Example:-
+}
+```
+
+### Weather
````
Weather should be part of the conditionals object, no?

BoogieMonster1O1 replied:
It is

AMereBagatelle replied:
Oh, I see now what you did.  Looks good now that I am looking at it as a not-git diff

AMereBagatelle commented on docs/schema-v2.md:
```diff
@@ -69,92 +195,158 @@ Example:-
+|         Name        |          Datatype         |                          Description                         |      Required      | Default value |
+|:-------------------:|:-------------------------:|:------------------------------------------------------------:|:------------------:|:-------------:|
+| `animationTextures` | Array of Textures objects | Specifies the list of textures to be used for each direction | :white_check_mark: |       -       |
+| `fps`               | Floating Point            | Specifies the number of frames to be rendered per second     | :white_check_mark: |       -       |
```
Maybe "Specifies the speed of the animation, in frames per second"? (for description of "fps")

AMereBagatelle replied:
On second thought, could be improved later but looks fine for now

AMereBagatelle reviewed (changes requested):
I think these were just missed or I can't read git diffs correctly
